### ISSUE TITLE
Inquiry on Analysis Duration with ANIb and ANIm for Streptomyces Genomes

### ISSUE THREAD

**Issue by ZiliaMR** [widdowquinn/pyani#420]
Dear Pyani Team,

I hope this message finds you well. I am currently conducting two separate analyses on a dataset comprising 310 genomes of Streptomyces, each with an approximate genome size of 6-8 Mb. For these analyses, I am utilizing the ANIb and ANIm methods, respectively. The computational resources allocated for this task include an 8-core processor with 49 GB of RAM.

Given the specifics of the dataset and the computational resources, I would greatly appreciate any insights or estimates you could provide regarding the expected duration for each analysis. 

Thank you for your assistance and time.
:)

**Comment by widdowquinn** [comment]
In general, though - assuming 1s per pairwise comparison, you can calculate an anticipated time for that of $n^2 / 3600$ hours, for $n$ genomes. There are other overheads, though - updating the database, parsing output files, etc. Also, ANIb takes considerably longer than ANIm.

The current GitHub version 0.3+ has a working fastANI implementation, which would speed things up at the cost of providing an ANI estimate, rather than an ANI calculation.

**Comment by kiepczi** [comment]
Hi @ZiliaMR. I have run aniM (pyANI version 0.3+) on 300 _Streptomyces_ genomes on a computer cluster with 40 cores and 192GB RAM, which took approximately 4 hours.

In comparison, I also ran much smaller analyses of a maximum of 70 _Streptomyces_ genomes on my local computer with 8 cores and 32GB RAM, which took around 3 hours. However, I ran this analysis overnight to limit background processes running on the computer, so there was less competition for the CPU.

Estimating an exact running time for a job is not an easy task, and it can depend on many factors, just as @widdowquinn said. His suggestion of estimating the running job is probably a good point to start. I also hope that my experiences will give a little bit more idea of how long it could take.
